src/main_app/components/SignUp/index.js

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react'
+import { loader } from 'graphql.macro'
+import { Formik, Field, Form, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
+import { useMutation, Mutation } from '@apollo/client'
+
+import client from 'services/graphql/client'
+
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms))
+
+const exampleMutation = loader('../../graphql/createUser.graphql')
+
+const SignupSchema = Yup.object().shape({
+  firstName: Yup.string()
+    .min(4, 'Too Short!')
+    .max(50, 'Too Long!')
+    .required('Required'),
+  lastName: Yup.string()
+    .min(2, 'Too Short!')
+    .max(50, 'Too Long!')
+    .required('Required'),
+  email: Yup.string().email('Invalid email').required('Required'),
+  password: Yup.string()
+    .min(8, 'Too Short!')
+    .max(12, 'Too Long!')
+    .required('Required')
+})
+
+const SignUp = () => {
+  const [state, setState] = useState ({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: ""
+  })
+  const { loading, error, data } = useMutation(exampleMutation, { client })
+  console.log(loading, error, data)
+
+  return (
+  <div>
+    <h1>Sign Up</h1>
+    <Formik
+      initialValues={{
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: ''
+      }}
+      validationSchema={SignupSchema}
+      onSubmit={async (values, actions) => {
+        await sleep(500)
+        alert(JSON.stringify(values, null, 2))
+        actions.setSubmitting(false)
+      }}
+    >
+      {({ isSubmitting, errors, touched }) => (
+        <Form onSubmit={e => e.preventDefault()}>
+
+          <label htmlFor="firstName">First Name</label>
+          <Field name="firstName" value={state.firstName} onChange={e => setState({ firstName: e.target.value })} />
```

Comment:
Don't handle Field values manually, let Formik do this for you.
```suggestion
          <Field name="firstName" />
```